syz-fuzzer/machine_info.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2020 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"strings"
+)
+
+func CollectMachineInfo() ([]byte, error) {
+	if runtime.GOOS != "linux" {
+		return nil, nil
+	}
+
+	type machineInfoFunc struct {
+		name string
+		fn   func() ([]byte, error)
+	}
+
+	allMachineInfo := []machineInfoFunc{
+		{"CPU Info", readCPUInfo},
+		{"KVM", readKVMInfo},
+	}
+
+	buffer := new(bytes.Buffer)
+
+	for _, pair := range allMachineInfo {
+		fmt.Fprintf(buffer, "[%s]\n", pair.name)
+		data, err := pair.fn()
```

Comment:
Rather than copying buffers multiple times, can we pass `buffer` to `fn` and write directly to it?